src/core/gui/sidebar/previews/page/SidebarPreviewPages.cpp

```diff
@@ -291,3 +297,11 @@ void SidebarPreviewPages::pageSelected(size_t page) {
 void SidebarPreviewPages::openPreviewContextMenu() {
     gtk_menu_popup(GTK_MENU(this->contextMenu), nullptr, nullptr, nullptr, nullptr, 3, gtk_get_current_event_time());
 }
+
+void SidebarPreviewPages::updateIndices() {
+    for (size_t i = 0; i < this->previews.size(); i++) {
+        SidebarPreviewBaseEntry* basePtr = this->previews.at(i).get();
+        SidebarPreviewPageEntry* preview = static_cast<SidebarPreviewPageEntry*>(basePtr);
+        preview->setIndex(static_cast<int>(i));
+    }
```

Comment:
Prefer iterator-based loops

```suggestion
    size_t index = 0;
    for (auto& preview: this->previews) {
        dynamic_cast<SidebarPreviewPageEntry*>(preview.get())->setIndex(index++));
    }
```